extensions/data-workspace/src/main.ts

```diff
@@ -8,12 +8,35 @@ import * as dataworkspace from 'dataworkspace';
 import { WorkspaceTreeDataProvider } from './common/workspaceTreeDataProvider';
 import { WorkspaceService } from './services/workspaceService';
 import { DataWorkspaceExtension } from './dataWorkspaceExtension';
+import { SelectProjectFileActionName } from './common/strings';
 
 export async function activate(context: vscode.ExtensionContext): Promise<dataworkspace.IExtension> {
 	const workspaceService = new WorkspaceService();
 	const workspaceTreeDataProvider = new WorkspaceTreeDataProvider(workspaceService);
 	context.subscriptions.push(vscode.window.registerTreeDataProvider('dataworkspace.views.main', workspaceTreeDataProvider));
-	context.subscriptions.push(vscode.commands.registerCommand('projects.addProject', () => {
+	context.subscriptions.push(vscode.commands.registerCommand('projects.addProject', async () => {
+		// To Sakshi - You can replace the implementation with your complete dialog implementation
+		// but all the code here should be reusable by you
+		if (vscode.workspace.workspaceFile) {
+			const filter: { [name: string]: string[] } = {};
+			const projectTypes = await workspaceService.getAllProjectTypes();
+			projectTypes.forEach(type => {
+				filter[type.displayName] = projectTypes.map(projectType => projectType.projectFileExtension);
+			});
+			let fileUris = await vscode.window.showOpenDialog({
+				canSelectFiles: true,
+				canSelectFolders: false,
+				canSelectMany: false,
```

Comment:
can opening multiple projects be allowed?

Reply:
that is TBD, for now, one at a time.

Reply:
For multiple projects, is there a way to have two sqlproj files in one folder? Or the expectation would be to populate it manually (instead of using the dialog)?